### PR TITLE
Feature payments

### PR DESCRIPTION
**Pull request by Bruh3509**
payments CRUD + connections between users and payments

### CODE REVIEW

File: flyway/sql/V5.0.16.08.2024.1344__create_table_payment.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+CREATE TABLE payment (
+    id BIGINT AUTO_INCREMENT NOT NULL PRIMARY KEY,
+    user_id BIGINT,
+    session_id BIGINT,
+    status VARCHAR(255),
+    url VARCHAR(255),
+    payment_amount INT,
```
Comment (Mishgan55) [inline flyway/sql/V5.0.16.08.2024.1344__create_table_payment.sql:7]: numeric

File: src/main/java/com/demo/cars/dto/PaymentDto.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import lombok.NoArgsConstructor;
+import lombok.experimental.FieldDefaults;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class PaymentDto {
+    Long userId;
+    Long sessionId; // ? - do we need these field
```
Comment (Mishgan55) [inline src/main/java/com/demo/cars/dto/PaymentDto.java:15]: maybe in a future, as far as i remember it is for Stripe API

File: src/main/java/com/demo/cars/entity/User.java [hunk 3]
```diff
@@ -1,24 +1,26 @@
...
 
 import java.sql.Timestamp;
+import java.util.LinkedList;
+import java.util.List;
 
 @Entity
 @Table(name = "_user")
 @Data
 @NoArgsConstructor
-@AllArgsConstructor
```
Comment (Mishgan55) [inline src/main/java/com/demo/cars/entity/User.java:21]: @EqualsAndHashCode(doNotUseGetters = true, onlyExplicitlyIncluded = true, callSuper = true)
@ToString(doNotUseGetters = true, onlyExplicitlyIncluded = true)

File: src/main/java/com/demo/cars/entity/User.java [hunk 4]
```diff
@@ -41,4 +43,27 @@ public class User {
 
     @Column(unique = true)
     String drivingLicenseId;
+
+    @OneToMany(mappedBy = "user", cascade = CascadeType.ALL)
+    List<Payment> payments = new LinkedList<>();
+
+    public User(
```
Comment (Mishgan55) [inline src/main/java/com/demo/cars/entity/User.java:54]: try to find another solution

File: flyway/sql/V5.0.16.08.2024.1344__create_table_payment.sql (hunk 1 above)
Comment (Bruh3509) [reply flyway/sql/V5.0.16.08.2024.1344__create_table_payment.sql:7]: So as I understand I need to change fields type to Double?

File: src/main/java/com/demo/cars/entity/User.java (hunk 3 above)
Comment (Bruh3509) [reply src/main/java/com/demo/cars/entity/User.java:21]: Should I add these annotations on every entity?

File: src/main/java/com/demo/cars/entity/User.java (hunk 4 above)
Comment (Bruh3509) [reply src/main/java/com/demo/cars/entity/User.java:54]: I tried earlier, and actually have no idea how to avoid these constructor

Comment (Bruh3509) [reply src/main/java/com/demo/cars/entity/User.java:54]: Maybe to user @NonNull annotation with @RequiredArgsConstructor, but I do not sure it's a good practice

Comment (Bruh3509) [reply src/main/java/com/demo/cars/entity/User.java:54]: Or I can make payments final -> so it won't be in @allargsconstructor. But I'm not sure that JPA do not change this list

**Review by Mishgan55 — APPROVED** [review]

**Pull request merged**